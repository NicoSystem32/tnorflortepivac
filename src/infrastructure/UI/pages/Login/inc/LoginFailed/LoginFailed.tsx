import { Link } from 'react-router-dom'

// styled components
import {
    LoginFailedContainer,
    TitleLoginContent,
    CustomerServiceContent,
    LoginTextInfo,
} from './loginFailed-styles'

// custom hooks
import { useLogin, useLoginState } from '../../hooks'

// Images
import { LoginPNG } from '../../../../utils/getImages'

// Icons
import { ArrowSVG } from '../../../../utils/getIcons'

const LoginFailed = (): JSX.Element => {
    // initial declarations
    const { loginMessages } = useLogin()
    const [{ feedback }] = useLoginState()

    const [title, message] = feedback.message.split('|')

    return (
        <LoginFailedContainer>
            <img src={LoginPNG} alt="" className="img-top" />
            <TitleLoginContent>
                <h3 className="title-login-failed">{title}</h3>
            </TitleLoginContent>

            {message && <LoginTextInfo>{message}</LoginTextInfo>}

            <LoginTextInfo>
                {loginMessages.find((m) => m.name === 'comunicateServicioAsociado')?.text}
            </LoginTextInfo>
            <CustomerServiceContent>
                <img src={ArrowSVG} alt="right arrow" />
                <Link id="support-lf" to="/support">
                    {loginMessages.find((m) => m.name === 'tituloServicioAsociado')?.text}
                </Link>
            </CustomerServiceContent>
        </LoginFailedContainer>
    )
}
export default LoginFailed
