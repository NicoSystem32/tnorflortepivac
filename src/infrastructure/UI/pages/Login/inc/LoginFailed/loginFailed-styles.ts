import tw, { styled } from 'twin.macro'

// styled components
import { StyledLoginForm } from '../LoginForm/loginForm-styles'

export const LoginFailedContainer = styled(StyledLoginForm)``

export const TitleLoginContent = styled.div`
    ${tw`pt-13 px-[5%] mb-12.5`}

    .title-login-failed {
        ${tw`font-montserrat font-semibold text-2xl text-black`}
    }
`

export const LoginTextInfo = styled.p`
    ${tw`font-montserrat font-normal text-base text-gray-custom-700 py-0 px-[5%] mb-3.5`}
`

export const CustomerServiceContent = styled.div`
    ${tw`flex pt-4 pb-0 px-[5%] gap-2.5`}

    a {
        ${tw`font-montserrat font-normal text-base text-dominant-dark no-underline`}
    }
`
