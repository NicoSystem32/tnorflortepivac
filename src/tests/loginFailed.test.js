import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import LoginFailed from '../infrastructure/UI/components/LoginFailed/loginFailed'

describe('Pruebas en <LoginFailed />', () => {
    const wrapper = shallow(<LoginFailed />)

    test('debe de mostrar <LoginFailed /> correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#support-lf').prop('href')).toBe('/support')
    })
})
