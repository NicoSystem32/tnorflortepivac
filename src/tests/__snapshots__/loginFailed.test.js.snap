// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginFailed /> debe de mostrar <LoginFailed /> correctamente 1`] = `
<div
  className="content-global-login"
>
  <img
    className="points-img"
    src="points.png"
    srcSet="points.svg"
  />
  <div
    className="content-global-init"
  >
    <Form
      className="form-login-failed failed-step1"
    >
      <img
        alt=""
        className="form-img"
        src="login.png"
      />
      <div
        className="content-title-form"
      >
        <p
          className="title-form-failed"
        >
          Parece que tenemos problemas con tu usuario
        </p>
      </div>
      <FormLabel
        className="group-label label-info"
        column={false}
        visuallyHidden={false}
      />
      <FormLabel
        className="group-label label-info-second"
        column={false}
        visuallyHidden={false}
      >
        Para continuar comun√≠cate con servicio al asociado
      </FormLabel>
      <div
        className="customer-service"
        id="content-customer-service"
      >
        <img
          alt=""
          className=""
          src="arrow-right.png"
          srcSet="arrow-right.svg"
        />
        <a
          href="/support"
          id="support-lf"
        >
          Contacta a servicio al asociado
        </a>
      </div>
    </Form>
  </div>
  <img
    alt=""
    className="elipse-img"
    src="elipse.png"
  />
  <img
    alt=""
    className="elipse-img-desk elipse-failed"
    src="elipse-desk.png"
  />
  <img
    alt=""
    className="networks-img failed"
    src="networks.png"
  />
  <img
    alt=""
    className="women-img"
    src="women.png"
  />
</div>
`;
